src/test/common/test_lockdep.cc

```diff
@@ -1,20 +1,45 @@
 // -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
 // vim: ts=8 sw=2 smarttab
 
-#include "common/ceph_context.h"
-#include "include/util.h"
 #include "gtest/gtest.h"
+
+#include "common/ceph_argparse.h"
+#include "common/ceph_context.h"
 #include "common/ceph_mutex.h"
-#include "global/global_context.h"
-#include "global/global_init.h"
+#include "common/common_init.h"
 #include "common/lockdep.h"
+#include "include/util.h"
 #include "include/coredumpctl.h"
 #include "log/Log.h"
 
-TEST(lockdep, abba)
+class lockdep : public ::testing::Test
 {
-  ASSERT_TRUE(g_lockdep);
+protected:
+  void SetUp() override {
+#ifdef CEPH_DEBUG_MUTEX
+    GTEST_SKIP() << "WARNING: CEPH_DEBUG_MUTEX is not defined, lockdep will not work";
```

Comment:
"CEPH_DEBUG_MUTEX is not defined" would seem to indicate `#ifndef` was intended.

see https://tracker.ceph.com/issues/46267

Reply:
@smithfarm  thanks Nathan, should be fixed by #35856